### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
See commits. This reorganized the entire documentation to be per audience, and adds internal docs (thanks to @hfiguiere).

### CODE REVIEW

File: doc/pull-requests.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Submitting Pull Requests
+========================
+
+XDG Desktop Portal uses GitHub pull requests to review contributions. Please be
+kind and patient as code reviews can be long and be minutious.
+
+Before developing features or fixing bugs, please make sure you are have done
```
Comment (swick) [inline doc/pull-requests.rst:7]: s/are//

File: doc/convenience-libraries.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+* `ASHPD <https://bilelmoussaoui.github.io/ashpd/ashpd/>`_: crate that provides
+  the APIs to interact with portals in idiomatic **Rust**. It has support for
+  GTK4, direct X11 windows, and direct Wayland surfaces.
+* `libportal <https://github.com/flatpak/libportal/>`_: small **C** library that
+  provides a GObject API to interact with portals. It provides language bindings
+  to a variety of other languages, such as **Python**, **JavaScript**, **Vala**,
+  and more. It has support for GTK3, GTK4, Qt 5, and Qt 6.
+* `xdg_desktop_portal <https://pub.dev/packages/xdg_desktop_portal>`_: a native
+  package to interact with portals in native **Dart** and **Flutter**.
```
Comment (hfiguiere) [inline doc/convenience-libraries.rst:16]:
```suggestion
* `ASHPD <https://bilelmoussaoui.github.io/ashpd/ashpd/>`_: a **Rust** crate that
  provides the APIs to interact with portals in idiomatic Rust. It has support for
  GTK4, direct X11 windows, and direct Wayland surfaces.
* `libportal <https://github.com/flatpak/libportal/>`_: small **C** library that
  provides a GObject API to interact with portals. It provides language bindings
  to a variety of other languages, such as **Python**, **JavaScript**, **Vala**,
  and more. It has support for GTK3, GTK4, Qt 5, and Qt 6.
* `xdg_desktop_portal <https://pub.dev/packages/xdg_desktop_portal>`_: a **Flutter** 
  native package to interact with portals in native **Dart** and Flutter.
```

File: doc/index.rst [hunk 3]
```diff
@@ -27,27 +27,21 @@ Content Overview
...
-  interact with the host system.
-* :doc:`Portal backends <backends>`: interfaces and configuration files that
-  **desktop developers** can implement and install in order to write a portal
-  backend.
-* :doc:`Background apps monitoring <introduction>`: service that provides
-  information about apps running in background to **desktop developers**.
-
-XDG Desktop Portal backends are selected and can be configured by using one or more
-configuration files. :doc:`Read more about them here <portals.conf>`.
+* :doc:`App Development <for-app-developers>`: portal APIs that **apps** can use
```
Comment (hfiguiere) [inline doc/index.rst:32]:
```suggestion
* :doc:`App Development <for-app-developers>`: portal APIs that **applications** can use
```

File: doc/writing-a-new-backend.rst [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+APIs and resources specific to the desktop environment they are running. For
+example, the KDE backend may use KDE-specific technologies, toolkit, and APIs.
+
+For a new portal backend (let's call it ``foo``) to be discoverable by XDG
+Desktop Portal, the following steps are necessary:
+
+* Implement one or more :doc:`backend D-Bus interfaces <impl-dbus-interfaces>`
+  in an executable. This executable must be `D-Bus activatable
+  <https://specifications.freedesktop.org/desktop-entry-spec/1.1/ar01s07.html>`_.
+* Install ``foo.portal`` file under `{DATADIR}/xdg-desktop-portal/portals`.
```
Comment (hfiguiere) [inline doc/writing-a-new-backend.rst:14]:
```suggestion
* Install ``foo.portal`` file under ``{DATADIR}/xdg-desktop-portal/portals``.
```

File: doc/writing-a-new-backend.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  Usually, ``{DATADIR}`` is `/usr/share`. The syntax of the portal file is
```
Comment (hfiguiere) [inline doc/writing-a-new-backend.rst:15]:
```suggestion
  Usually, ``{DATADIR}`` is ``/usr/share``. The syntax of the portal file is
```

**Comment by hfiguiere** [comment]
Looking great !

File: doc/index.rst (hunk 3 above)
Comment (GeorgesStavracas) [reply doc/index.rst:32]: We've been explicitly using `app` instead of `application` these days

Comment (hfiguiere) [reply doc/index.rst:32]: like using `crypto` for `cryptography` ? Clear language is important. The proper word is application.

Comment (swick) [reply doc/index.rst:32]: App is the short form of application, crypto was the short form of cryptographic before it was taken over by blockchain bullshit. Nothing wrong with app.

**Pull request merged**